File: resources/token.go
package resources

import "context"

// Token handles token-related API operations
type Token struct {
	client ClientInterface
}

// NewToken creates a new Token resource client
func NewToken(client ClientInterface) *Token {
	return &Token{client: client}
}

// RevokeRequest represents a token revocation request
type RevokeRequest struct {
	Token         string  `json:"token"`
	TokenTypeHint *string `json:"token_type_hint,omitempty"`
}

// UserInfo represents information about the authenticated user/token
type UserInfo struct {
	Iat      int64   `json:"iat"`                 // Issued at timestamp (Unix epoch seconds)
	Scope    string  `json:"scope"`               // Granted scopes
	Sub      string  `json:"sub"`                 // Subject identifier (token ID)
	TokenUse string  `json:"token_use"`           // Token type
	Aud      *string `json:"aud,omitempty"`       // Audience (client ID)
	ClientID *string `json:"client_id,omitempty"` // Client identifier
	Exp      *int64  `json:"exp,omitempty"`       // Expiration timestamp (Unix epoch seconds)
}

// Info returns information about the authenticated user/token
func (t *Token) Info(ctx context.Context) (*UserInfo, error) {
	var result UserInfo
	err := t.client.DoRequest(ctx, "GET", "/oauth/userinfo", nil, &result)
	if err != nil {
		return nil, err
	}
	return &result, nil
}

File: beeper-rust-sdk/src/resources/token.rs
use crate::client::BeeperDesktop;
use crate::error::Result;
use reqwest::Method;
use serde::{Deserialize, Serialize};

/// Token handles token-related API operations
#[derive(Debug, Clone)]
pub struct Token {
    client: BeeperDesktop,
}

impl Token {
    /// Create a new Token resource client
    pub fn new(client: BeeperDesktop) -> Self {
        Self { client }
    }

    /// Info returns information about the authenticated user/token
    pub async fn info(&self) -> Result<UserInfo> {
        self.client
            .do_request(Method::GET, "/oauth/userinfo", None::<&()>)
            .await
    }
}

/// RevokeRequest represents a token revocation request
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RevokeRequest {
    pub token: String,
    pub token_type_hint: Option<String>,
}

/// UserInfo represents information about the authenticated user/token
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UserInfo {
    /// Issued at timestamp (Unix epoch seconds)
    pub iat: i64,
    /// Granted scopes
    pub scope: String,
    /// Subject identifier (token ID)
    pub sub: String,
    /// Token type
    pub token_use: String,
    /// Audience (client ID)
    pub aud: Option<String>,
    /// Client identifier
    pub client_id: Option<String>,
    /// Expiration timestamp (Unix epoch seconds)
    pub exp: Option<i64>,
}